src/box/memtx_space.c

```diff
@@ -336,9 +337,9 @@ memtx_space_execute_replace(struct space *space, struct txn *txn,
 	struct memtx_space *memtx_space = (struct memtx_space *)space;
 	struct txn_stmt *stmt = txn_current_stmt(txn);
 	enum dup_replace_mode mode = dup_replace_mode(request->type);
-	stmt->new_tuple =
-		space->format->vtab.tuple_new(space->format, request->tuple,
-					      request->tuple_end);
+
+	stmt->new_tuple = tt_compress_tuple_new(space, request->tuple,
+						request->tuple_end);
```

Comment:
Can't you put compression logic invocation in `memtx_tuple_new` instead?

Reply:
According to our private discussion leave it as it is.